### PR TITLE
RPC DQM occupancy bugfix + cleanup

### PR DESCRIPTION
**Pull request by jhgoh**
This PR fixes problems in the RPC offline DQM.

Problem:
In the RPC summary plots (offline DQM), ring3/A,C shows very low occupancy in 2D plots (https://goo.gl/sWUTxU). This is not consistent with <Run2016 nor the online DQM. It also disagree with 1D distributions. Same symptoms are observed in release validation.

Observation:
We could reproduce the same problem with RelVal samples **with multi-threading**. 
The distribution becomes OK with single thread mode.
Affected histograms have custom labels in each axis, without calling setBinLabel() restores nominal shapes.

Solution:
Explicitly call TAxis::SetNoAlphanumeric() for the MonitorElements before setting bin labels.

image

image

Other changes:
We also added some changes for cleanup the codes and remove obsolete scripts.

@slezki @mileva @andresib

### CODE REVIEW

**Comment by dmitrijus** [comment]
+1

**Comment by dmitrijus** [comment]
Can you provide more details of this bug?

We had a similar issue in the past (root developers fixed it quickly),
I want to make sure it did not come back...

**Comment by jhgoh** [comment]
@dmitrijus what we found is that some of 2D plots with bin labels are affected, some entries are filled in different place. It does not happen for the single thread mode but happens if we turn on MT - online DQM is not affected, but offline & RelVal are affected.

The total number of entries were same, indicating no corruption but just FindBin() is working in unexpected way. 
Calling TAxis::SetNoAlphanumeric() before setting labels seem to be working, enforcing the axis behaviours to be unchanged during the histogram booking. This is reasonable for us because we are using (abusing?) axis labels for decoration. We still want to fill histograms not based on the bin "labels". 

For more details: From https://root.cern.ch/content/main-histogram-changes-root-6
> Histograms with labels/Alphanumeric axis
>
> In ROOT 6 we have introduced the notion of an alphanumeric axis. An axis is considered alphanumeric when all the bins have labels attached to them. In this case the numerical bin center values do not really make sense and one should fill the histogram with TH1::Fill(label) instead of TH1::Fill(x), where label is a string and x is a real number. 
> In the case of an alphanumeric axis, the histogram statistics (Mean, RMS, etc..) is not computed anymore, and their values are set to zero. The histograms with an alphanumeric axis will be merged by adding the bin contents for bins with the same labels. 
> If a user wants to force an axis to be not alphanumeric, although labels are present in every bin bin, he can call the function

File: DQM/RPCMonitorDigi/interface/utils.h [hunk 1]
```diff
@@ -13,6 +13,21 @@ namespace rpcdqm{
 
   enum RPCMeFLag{OCCUPANCY = 1, CLUSTERSIZE = 2, MULTIPLICITY =3, BX=4};
 
+  struct RPCMEHelper
+  {
+    static void setNoAlphanumeric(MonitorElement* myMe)
+    {
+      // Set no-alphanumeric flag to avoid malfunctioning in multithread environment.
+      TH2* h2 = nullptr;
```
Comment (davidlange6) [inline DQM/RPCMonitorDigi/interface/utils.h]: hi @dmitrijus @jhgoh - could you add a getTH2() method to MonitorElement to avoid this complexity. Thanks

**Comment by davidlange6** [comment]
otherwise, @dmitrijus - my old emails suggest that core dqm was going to call setnoalphanumeric to avoid this issue. Did that not happen?

File: DQM/RPCMonitorDigi/interface/utils.h (hunk 1 above)
Comment (jhgoh) [reply DQM/RPCMonitorDigi/interface/utils.h]: I can simply use getTH1() since TH2 inherits from TH1.

**Comment by jhgoh** [comment]
code is updated to address @davidlange6 's comment.

**Comment by dmitrijus** [comment]
@davidlange6 I find that suggestion was a just temporary "workaround".
DQM has nothing against alphanumeric histograms.

The issue back then was that alphanumeric histograms could be implicitly extended via fill / findBin. (axis extension is problematic for DQM, since we can no longer use TH1::Add()).

And it was fixed by this commit, https://github.com/root-project/root/commit/1796bf22743b9b37bedd29664b718b5cdf866d16

This RPC DQM bug looks very similar to how an axis extension would look like.
There a big mess will all the root TH1 flags and implicit toggles of them, I am still investigating.

Either way, this PR should be merged. If a subsystem/module wants to explicitly avoid having alphanumeric histograms, they should disable it just like it was done in this PR.

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Comment by jhgoh** [comment]
@dmitrijus should I prepare backport(s) to apply this bugfix as well?

**Comment by dmitrijus** [comment]
@jhgoh That would welcome, yes :)

**Comment by jhgoh** [comment]
93X backport is created. Better to prepare one more for 92X?

**Pull request merged**